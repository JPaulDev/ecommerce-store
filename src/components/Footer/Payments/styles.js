import styled from 'styled-components';

export const Container = styled.div`
  grid-column: 1 / 3;

  @media screen and (min-width: 1050px) {
    grid-column: initial;
    margin-right: 50px;
  }
`;

export const List = styled.ul`
  background-color: var(--color-grey-11);
  display: flex;
  flex-wrap: wrap;
  gap: 10px;
  justify-content: center;
  list-style: none;
  padding: 10px;

  @media screen and (min-width: 1050px) {
    max-width: 135px;
    background: none;
    margin: 0 auto 20px;
    padding: 12px 0;
  }
`;

export const ListItem = styled.li`
  background-color: var(--color-grey-6);
  border-radius: 6px;
  display: grid;
  height: 37px;
  place-items: center;
  width: 60px;

  @media screen and (min-width: 1050px) {
    background-color: var(--color-grey-11);
  }
`;
