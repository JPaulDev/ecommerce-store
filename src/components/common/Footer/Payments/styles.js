import styled, { css } from 'styled-components';

const flexStyles = css`
  display: flex;
  justify-content: center;
`;

export const Container = styled.div`
  grid-column: 1 / 3;

  @media screen and (min-width: 1050px) {
    grid-column: initial;
    margin-right: 50px;
  }
`;

export const List = styled.ul`
  background-color: var(--color-grey-11);
  flex-wrap: wrap;
  gap: 10px;
  list-style: none;
  padding: 10px;
  ${flexStyles}

  @media screen and (min-width: 1050px) {
    max-width: 135px;
    background: none;
    margin: 0 auto 20px;
    padding: 12px 0;
  }
`;

export const ListItem = styled.li`
  align-items: center;
  background-color: var(--color-grey-6);
  border-radius: 6px;
  height: 37px;
  width: 60px;
  ${flexStyles}

  @media screen and (min-width: 1050px) {
    background-color: var(--color-grey-11);
  }
`;
