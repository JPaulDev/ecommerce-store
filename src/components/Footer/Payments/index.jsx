import useMediaQuery from '../../../hooks/useMediaQuery';
import { ReactComponent as Mastercard } from '../../../assets/svg/mastercard.svg';
import { ReactComponent as Visa } from '../../../assets/svg/visa.svg';
import { ReactComponent as Electron } from '../../../assets/svg/electron.svg';
import { ReactComponent as AmazonPay } from '../../../assets/svg/amazon-pay.svg';
import { ReactComponent as Maestro } from '../../../assets/svg/maestro.svg';
import { ReactComponent as Paypal } from '../../../assets/svg/paypal.svg';
import { ReactComponent as AmericanExpress } from '../../../assets/svg/american-express.svg';
import { ReactComponent as Bitcoin } from '../../../assets/svg/bitcoin.svg';
import { Heading } from '../styles';
import * as Styled from './styles';

const logos = [
  <Visa width={48} title="Pay with Visa" aria-label="Visa" />,
  <Mastercard width={53} title="Pay with Mastercard" aria-label="Mastercard" />,
  <Electron
    width={48}
    title="Pay with Visa Electron"
    aria-label="Visa Electron"
  />,
  <AmazonPay width={54} title="Pay with Amazon Pay" aria-label="Amazon Pay" />,
  <Maestro width={47} title="Pay with Maestro" aria-label="Maestro" />,
  <Paypal width={54} title="Pay with PayPal" aria-label="PayPal" />,
  <AmericanExpress
    width={50}
    title="Pay with American Express"
    aria-label="American Express"
  />,
  <Bitcoin width={50} title="Pay with Bitcoin" aria-label="Bitcoin" />,
];

export default function Payments() {
  const isMatch = useMediaQuery('(min-width: 1050px)');

  return (
    <Styled.Container>
      {isMatch && <Heading>Secure Payments</Heading>}
      <Styled.List>
        {logos.map((Logo, index) => (
          <Styled.ListItem key={index}>{Logo}</Styled.ListItem>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
