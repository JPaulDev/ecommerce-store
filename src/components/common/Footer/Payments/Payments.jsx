import { useMediaQuery } from '../../../../hooks';
import {
  Visa,
  Mastercard,
  Electron,
  AmazonPay,
  Maestro,
  PayPal,
  AmericanExpress,
  Bitcoin,
} from '../../../icons';
import { Heading } from '../styles';
import * as Styled from './styles';

const logos = [
  {
    title: 'Pay with Visa',
    icon: <Visa width={48} height={48} aria-label="Visa" />,
  },
  {
    title: 'Pay with Mastercard',
    icon: <Mastercard width={53} height={53} aria-label="Mastercard" />,
  },
  {
    title: 'Pay with Visa Electron',
    icon: <Electron width={48} height={48} aria-label="Visa Electron" />,
  },
  {
    title: 'Pay with Amazon Pay',
    icon: <AmazonPay width={54} height={18} aria-label="Amazon Pay" />,
  },
  {
    title: 'Pay with Maestro',
    icon: <Maestro width={47} height={47} aria-label="Maestro" />,
  },
  {
    title: 'Pay with PayPal',
    icon: <PayPal width={54} height={13} aria-label="PayPal" />,
  },
  {
    title: 'Pay with American Express',
    icon: (
      <AmericanExpress width={50} height={18} aria-label="American Express" />
    ),
  },
  {
    title: 'Pay with Bitcoin',
    icon: <Bitcoin width={50} height={10} aria-label="Bitcoin" />,
  },
];

export default function Payments() {
  const isMatch = useMediaQuery('(min-width: 1050px)');

  return (
    <Styled.Container>
      {isMatch && <Heading>Secure Payments</Heading>}
      <Styled.List>
        {logos.map((item) => (
          <Styled.ListItem key={item.title} title={item.title}>
            {item.icon}
          </Styled.ListItem>
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
