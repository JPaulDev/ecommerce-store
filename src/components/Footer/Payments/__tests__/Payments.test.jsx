import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Payments from '../index';

describe('Payments component', () => {
  it('renders a list of accepted payment methods', () => {
    const tree = renderer.create(<Payments />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Payments component at different screen widths', () => {
  it('renders a heading at 1050px and above', () => {
    window.resizeTo(1050, 768);
    render(<Payments />);

    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('does not render a heading at 1049px and below', () => {
    window.resizeTo(1049, 768);
    render(<Payments />);

    expect(screen.queryByRole('heading')).not.toBeInTheDocument();
  });
});
