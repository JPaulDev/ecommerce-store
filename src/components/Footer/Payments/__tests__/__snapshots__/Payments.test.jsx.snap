// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payments component renders a list of accepted payment methods 1`] = `
<div
  className="sc-jSMfEi iQyKUy"
>
  <ul
    className="sc-gKXOVf bFfhea"
  >
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="Visa"
        title="Pay with Visa"
        width={48}
      >
        visa.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="Mastercard"
        title="Pay with Mastercard"
        width={53}
      >
        mastercard.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="Visa Electron"
        title="Pay with Visa Electron"
        width={48}
      >
        electron.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="Amazon Pay"
        title="Pay with Amazon Pay"
        width={54}
      >
        amazon-pay.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="Maestro"
        title="Pay with Maestro"
        width={47}
      >
        maestro.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="PayPal"
        title="Pay with PayPal"
        width={54}
      >
        paypal.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="American Express"
        title="Pay with American Express"
        width={50}
      >
        american-express.svg
      </svg>
    </li>
    <li
      className="sc-iBkjds lcBsOn"
    >
      <svg
        aria-label="Bitcoin"
        title="Pay with Bitcoin"
        width={50}
      >
        bitcoin.svg
      </svg>
    </li>
  </ul>
</div>
`;
